lib/puppet/resource.rb

```diff
@@ -385,104 +384,6 @@ def name
     [ type, title ].join('/')
   end
 
-  def missing_arguments
-    resource_type.arguments.select do |param, default|
-      the_param = parameters[param.to_sym]
-      the_param.nil? || the_param.value.nil? || the_param.value == :undef
-    end
-  end
-  private :missing_arguments
-
-  # Consult external data bindings for class parameter values which must be
-  # namespaced in the backend.
-  #
-  # Example:
-  #
-  #   class foo($port=0){ ... }
-  #
-  # We make a request to the backend for the key 'foo::port' not 'foo'
-  #
-  def lookup_external_default_for(param, scope)
```

Comment:
No it does not. The lookup_external_default_for is a :private method (look beneath the method)